SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -366,3 +378,304 @@ void CIncNSSolver::BC_ConjugateHeat_Interface(CGeometry *geometry, CSolver **sol
+
+          counter = 0; diff = 1.0;
+          U_Tau = sqrt(WallShearStress/Density_Wall);
+          Y_Plus = 0.0; // to avoid warning
+
+          su2double Y_Plus_Start = Density_Wall * U_Tau * WallDistMod / Lam_Visc_Wall;
+
+          /*--- Automatic switch off when y+ < 5 according to Nichols & Nelson (2004) ---*/
+
+          if (Y_Plus_Start < 5.0) {
+            continue;
+          }
+
+          while (fabs(diff) > tol) {
+
+            /*--- Friction velocity and u+ ---*/
+
+            U_Plus = VelTangMod/U_Tau;
+
+            /*--- Gamma, Beta, Q, and Phi, defined by Nichols & Nelson (2004) page 1110 ---*/
+
+            Gam  = Recovery*U_Tau*U_Tau/(2.0*Cp*T_Wall);
+            Beta = q_w*Lam_Visc_Wall/(Density_Wall*T_Wall*Conductivity_Wall*U_Tau); // TODO: nonzero heatflux needs validation case
+            Q    = sqrt(Beta*Beta + 4.0*Gam);
+            Phi  = asin(-1.0*Beta/Q);
+
+            /*--- Y+ defined by White & Christoph (compressibility and heat transfer) negative value for (2.0*Gam*U_Plus - Beta)/Q ---*/
+
+            Y_Plus_White = exp((kappa/sqrt(Gam))*(asin((2.0*Gam*U_Plus - Beta)/Q) - Phi))*exp(-1.0*kappa*B);
+
+            /*--- Spalding's universal form for the BL velocity with the
+             outer velocity form of White & Christoph above. ---*/
+
+            Y_Plus = U_Plus + Y_Plus_White - (exp(-1.0*kappa*B)*
+                                              (1.0 + kappa*U_Plus + kappa*kappa*U_Plus*U_Plus/2.0 +
+                                               kappa*kappa*kappa*U_Plus*U_Plus*U_Plus/6.0));
+
+            su2double dypw_dyp = 2.0*Y_Plus_White*(kappa*sqrt(Gam)/Q)*sqrt(1.0 - pow(2.0*Gam*U_Plus - Beta,2.0)/(Q*Q));
+            Eddy_Visc = Lam_Visc_Wall*(1.0 + dypw_dyp - kappa*exp(-1.0*kappa*B)*
+                                             (1.0 + kappa*U_Plus + kappa*kappa*U_Plus*U_Plus/2.0)
+                                             - Lam_Visc_Normal/Lam_Visc_Wall);
+            Eddy_Visc = max(1.0e-6, Eddy_Visc);
+
+            /* --- Define function for Newton method to zero --- */
+
+            diff = (Density_Wall * U_Tau * WallDistMod / Lam_Visc_Wall) - Y_Plus;
+
+            /* --- Gradient of function defined above --- */
+
+            grad_diff = Density_Wall * WallDistMod / Lam_Visc_Wall + VelTangMod / (U_Tau * U_Tau) +
+                      kappa /(U_Tau * sqrt(Gam)) * asin(U_Plus * sqrt(Gam)) * Y_Plus_White -
+                      exp(-1.0 * B * kappa) * (0.5 * pow(VelTangMod * kappa / U_Tau, 3) +
+                      pow(VelTangMod * kappa / U_Tau, 2) + VelTangMod * kappa / U_Tau) / U_Tau;
+
+            /* --- Newton Step --- */
+
+            U_Tau = U_Tau - 0.5*(diff / grad_diff);
+
+            counter++;
+
+            /* --- if we do not converge, then something serious is going on and we stop --- */
+
+            if (counter == max_iter) {
+              SU2_MPI::Error("Y+ did not converge within the max number of iterations!",CURRENT_FUNCTION);
+            }
```

Comment:
Maybe we can be a bit more forgiving to avoid blowing up during initial solution transients, e.g. count the number of failed points and report at the end of the function without throwing an error.

Reply:
yes, maybe just output a "warning: XX points did not converge"